### PR TITLE
Template Sendgrid - Use OrgEmailData instead of Organization and others fixes

### PR DESCRIPTION
**Pull request by KevinDurand974**
From issue: https://github.com/blockframes/blockframes/issues/7491

Changelog:
- Remove `Organization` and `Movie` from `EmailTemplateRequest`
- Rename `MailBucket` to `BucketEmailData`
- Check `templateIds.offer.created` who's seems to be removed: No removed, it's now `templateIds.offer.toAdmin`
- Fix a link into template `52467be7bfdd43c89bec2a1f3cca4374` to `mailto:`
- Standarize the syntax `pageUrl` & `pageURL` to `pageUrl` only and update all templates

### List of template who need to change the version:

- [ ] Prod Edited
`b928481ce938423dabb34e50c264d320` > **V1**
`81438bdf511b43cfa866ca63a45a02ae` > **V5**
`fc05a8cf5b1548ebae9ca44247a6c256` > **V7**
`6a0710945bc841ffb6955e3dc202704c` > **V4**
`496956508c64403393a6e5f5e2392e6c` > **V4**
`a4ee1970187e4fd8bef47a4008a2267a` > **V1**
`ac1dfcbf692d4ef8888ccad9b6938c8e` > **V1**
`4f067ef3abca42cc9d8dc25b57935ccc` > **V3** (Disabled at this time)
`c92f8799273f497cb2256eb6ed8baa78` > **V3** (Disabled at this time)
`38ee6949f4724b1c8796b8df9229f21d` > **V4** (Disabled at this time)

- [x] From #8355
`022cee72b5904d7386a838d6317054ad` > **V2**
`5c03ed5bde6d40768a44e3d1e1c95b67` > **V8**
`4e646560616b4030be5b8fdae4dfc50d` > **V8**
`95a9ae5824044c42ae90d804146c5080` > **V6**
`52467be7bfdd43c89bec2a1f3cca4374` > **V2**
`afa28cb76e034d9b969588db5933362f` > **V2**
`31ba12e3df9946bcb52fefca0c431240` > **V2**
`2e90ffb26f2e462ca574dccbc1e80ba2` > **V2**
`5826727a7741475cbe120a8ef50e3cc7` > **V2**
`a8098e7ac093457ba834397a4657daad` > **V2**
`13e08d0d89d94cc880a8f40b9880e683` > **V4**

### CODE REVIEW

File: apps/backend-functions/src/notification.ts [hunk 1]
```diff
@@ -461,7 +461,7 @@ async function sendScreeningRequested(recipient: User, notification: Notificatio
     toUser,
```
Comment (phpgeekfr) [inline apps/backend-functions/src/notification.ts:461]:
```suggestion
    getUserEmailData(recipient),
```

File: libs/model/src/lib/emails.ts [hunk 2]
```diff
@@ -97,16 +97,16 @@ export interface EmailTemplateRequest {
   to: string;
   templateId: string;
   data: {
-    org?: OrgEmailData | Organization; // @TODO #7491 template d-f45a08ce5be94e368f868579fa72afa8 uses Organization but it should use OrgEmailData instead
+    org?: OrgEmailData; // @TODO #7491 template d-f45a08ce5be94e368f868579fa72afa8 uses Organization but it should use OrgEmailData instead
```
Comment (phpgeekfr) [inline libs/model/src/lib/emails.ts:100]: remove the comment

File: libs/model/src/lib/bucket.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ export interface BucketContract {
 }
 
 //To be used for sending mails.
-export interface MailBucket {
+export interface BucketEmailData {
```
Comment (phpgeekfr) [inline libs/model/src/lib/bucket.ts:32]: move it to `libs/model/src/lib/emails.ts`

Comment (phpgeekfr) [reply libs/model/src/lib/bucket.ts:32]: and remove `export`

Comment (phpgeekfr) [reply libs/model/src/lib/bucket.ts:32]: please also remove export on `NegotiationEmailData` and `OfferEmailData`

File: libs/model/src/lib/emails.ts [hunk 4]
```diff
@@ -97,16 +97,16 @@ export interface EmailTemplateRequest {
...
-    bucket?: MailBucket;
+    pageUrl?: string;
+    bucket?: BucketEmailData;
     baseUrl?: string;
     date?: string;
-    movie?: MovieEmailData | Movie;
+    movie?: MovieEmailData;
     offer?: OfferEmailData;
     buyer?: UserEmailData;
     contract?: Contract;
```
Comment (phpgeekfr) [inline libs/model/src/lib/emails.ts:112]: you did not update `contract?: Contract;` but it looks like unused.

remove it from here and check if others can't also be removed.

you can remove an attribute and `npx nx build backend-functions` to check if it is used or not

Comment (phpgeekfr) [reply libs/model/src/lib/emails.ts:112]: `territories` seems to be in same case

Comment (KevinDurand974) [reply libs/model/src/lib/emails.ts:111]: `contract` and `territories` are used in 6 - 8 templates, the only things that was removed are `eventUrl` and `contractId` who're not used anymore

File: libs/model/src/lib/contract.ts [hunk 5]
```diff
@@ -3,7 +3,7 @@ import type { Media, Territory, ContractStatus } from './static';
 import { createMailTerm, Duration } from './terms';
 import { BucketContract } from './bucket';
 
-export function createMailContract(contract: BucketContract) {
+export function getContractEmailData(contract: BucketContract) {
   const formatter = new Intl.NumberFormat('en-US');
   const price = contract.price ? formatter.format(contract.price) : '';
```
Comment (phpgeekfr) [inline libs/model/src/lib/contract.ts:9]: this method should not return the contract entirely but just the required data for emails

File: libs/model/src/lib/contract.ts [hunk 6]
```diff
@@ -58,7 +58,7 @@ export interface Sale extends Contract {
   holdbacks: Holdback[];
 }
 
-export type MailContract = ReturnType<typeof createMailContract>;
+export type ContractEmailData = ReturnType<typeof getContractEmailData>;
```
Comment (phpgeekfr) [inline libs/model/src/lib/contract.ts:61]: this should be `libs/model/src/lib/emails.ts` like others
and should be like this (not `ReturnType<typeof getContractEmailData`):
```
export interface MovieEmailData {
  id: string;
  title: {
    international: string;
  };
}
```

File: libs/model/src/lib/emails.ts [hunk 7]
```diff
@@ -192,6 +197,16 @@ export function getBucketEmailData(bucket: Bucket): BucketEmailData {
...
 }
 
+export function getContractEmailData({ titleId, price, terms }: BucketContract): ContractEmailData {
+  const formatter = new Intl.NumberFormat('en-US');
+
+  return ({
+    titleId,
+    price: price ? formatter.format(price) : '',
+    terms: createMailTerm(terms)
+  });
```
Comment (phpgeekfr) [inline libs/model/src/lib/emails.ts:207]:
```suggestion
  return {
    titleId,
    price: price ? formatter.format(price) : '',
    terms: createMailTerm(terms)
  };
```

File: libs/model/src/lib/emails.ts [hunk 8]
```diff
@@ -183,6 +205,31 @@ export function getBucketEmailData(bucket: Bucket): MailBucket {
   };
 }
 
+export function getContractEmailData({ titleId, price, terms }: BucketContract): ContractEmailData {
+  const formatter = new Intl.NumberFormat('en-US');
```
Comment (phpgeekfr) [inline libs/model/src/lib/emails.ts:209]: `const formatter = new Intl.NumberFormat('en-US');` is used both in this file, you can create the const outside any function

**Comment by cams-goussale** [comment]
Tested on prod?

- [ ] `b928481ce938423dabb34e50c264d320` > **V1** update pageUrl and declineReason
- [ ] `81438bdf511b43cfa866ca63a45a02ae` > **V5** update pageUrl
- [ ] `fc05a8cf5b1548ebae9ca44247a6c256` > **V7** update pageUrl
- [ ] `6a0710945bc841ffb6955e3dc202704c` > **V4** update pageUrl
- [ ] `496956508c64403393a6e5f5e2392e6c` > **V4** update pageUrl
- [ ] `a4ee1970187e4fd8bef47a4008a2267a` > **V1** update pageUrl
- [ ] `ac1dfcbf692d4ef8888ccad9b6938c8e` > **V1** update pageUrl
- [ ] `4f067ef3abca42cc9d8dc25b57935ccc` > **V3** update pageUrl (Disabled at this time)
- [ ] `c92f8799273f497cb2256eb6ed8baa78` > **V3** update pageUrl (Disabled at this time)
- [ ] `38ee6949f4724b1c8796b8df9229f21d` > **V4** update pageUrl (Disabled at this time)

**Pull request merged**